### DISCUSSION TITLE
Support in place edit

### DISCUSSION THREAD

**Discussion by semberal** [ente-io/ente#578] [Enhancements]
Hi, when I edit a photo, I would like to have the option to replace the existing photo instead of creating a copy. When I need to rotate a bunch of photos, for example, I don't care about the originals anymore and it is a bit tedious to have to look up the originals and remove them.

**Comment by vishnukvmd** [comment]
Just to add some context, writing all of the metadata back into the updated file post edits is non-trivial for certain file formats (like HEIC).

We'd rather not get rid off the originals until there's a clear path forward where we can replicate the original metadata into the updated file.

**Comment by vishnukvmd** [comment]
Feedback from Discord: Ente could internally stack edited images on top of the originals, so that only the former is visible on the UI, and a customer has the option to revert to the original if / when they wish.

**Comment by C0smin** [comment]
Hi everyone, started using ente a couple of days ago, I think it's amazing. And I agree rotating an image should be a lot easier than it is now. On that note, while I understand that _writing all of the metadata back into the updated file post edits is non-trivial for certain file formats_ the edit image feature is probably working under the same limitations, in this case wouldn't this feature be only a _shortcut_ to the Edit -> Rotate -> Save functionality?
Also, this was not immediately evident to me, so any change I do to a file by what you are saying, it will never be close to the original as it is a _new_ file correct? (should be mentioned somewhere in the docs if it's not already, I'm very sorry if i missed it)

**Reply by vishnukvmd** [reply]
> wouldn't this feature be only a shortcut to the Edit -> Rotate -> Save functionality?

Sorry, did not get this.

> any change I do to a file by what you are saying, it will never be close to the original

We are creating a new file and writing the pixels to it, so yes, it is different from the original.

**Reply by C0smin** [reply]
Thank you for such a fast reply! 
> Sorry, did not get this.

The current feature set gives you the ability to:
* Edit picture
* Rotate left/right 90
* Save a copy to Ente
My suggestion is to capture this series of action within a single button.

> We are creating a new file and writing the pixels to it, so yes, it is different from the original.

Very good to know, in this case creating what I suggested above will not work, as you would be left with a _dangling_ image (the one you want to see but not where and how).

**Comment by samsonjs** [comment]
A bunch of these are related:

- #2535
- #4299
- #6742

This could all be addressed by having non-destructive edits that are synced with photos/videos. I would love that so hard.
